src/engine/QueryPlanner.cpp

```diff
@@ -865,44 +862,43 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::getDistinctRow(
       }
     }
     const std::vector<size_t>& resultSortedOn =
-        parent._qet->getRootOperation()->getResultSortedOn();
+        i._qet->getRootOperation()->getResultSortedOn();
     // check if the current result is sorted on all columns of the distinct
     // with the order of the sorting
     bool isSorted = resultSortedOn.size() >= keepIndices.size();
-    for (size_t i = 0; isSorted && i < keepIndices.size(); i++) {
-      isSorted = isSorted && resultSortedOn[i] == keepIndices[i];
+    for (size_t ii = 0; isSorted && ii < keepIndices.size(); ii++) {
+      isSorted = isSorted && resultSortedOn[ii] == keepIndices[ii];
```

Comment:
If you really want to use a different counting variable (is the `i` shadowing a global `i`), the use `j` or `k`, not `ii`